src/components/ChatEntry.js

```diff
@@ -1,22 +1,43 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
 
 const ChatEntry = (props) => {
+
+  const toggleLikeButton = () => {
+    props.updateLiked(props.id);
+  };
+
+  // Note: Tests for waves don't pass if I have the commented-out lines :,( 
+  // But if I don't use colorData, the tests will pass. 
+  // Just wanted to do the optional enhancement. It works in browser!
+
+  // const textColor = props.colorData[props.sender];
+
   return (
-    <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+    <div className={ props.id % 2 ? 'chat-entry local' : 'chat-entry remote' }>
+      <h2 className="entry-name">{props.sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
-      </section>
+        {/* <p className={ textColor }> {props.body} </p> */}
+        <p> {props.body} </p>
+        <p className="entry-time">
+          <TimeStamp time={props.timeStamp}></TimeStamp>
```

Comment:
[suggestion] You're allowed to use self-closing tags for cases like this
```suggestion
          <TimeStamp time={props.timeStamp} />
```